src/client/action/watch.rs

```diff
@@ -0,0 +1,350 @@
+    /// Starts a new [`ChangeStream`](change_stream/struct.ChangeStream.html) that receives events
+    /// for all changes in this collection. A
+    /// [`ChangeStream`](change_stream/struct.ChangeStream.html) cannot be started on system
+    /// collections.
+    ///
+    /// See the documentation [here](https://www.mongodb.com/docs/manual/changeStreams/) on change
+    /// streams.
+    ///
+    /// Change streams require either a "majority" read concern or no read concern. Anything else
+    /// will cause a server error.
+    pub fn watch(&self) -> Watch {
+        Watch::new(self.client(), self.namespace().into())
+    }
+}
+
+#[cfg(any(feature = "sync", feature = "tokio-sync"))]
+impl crate::sync::Client {
+    /// Starts a new [`ChangeStream`] that receives events for all changes in the cluster. The
+    /// stream does not observe changes from system collections or the "config", "local" or
+    /// "admin" databases. Note that this method (`watch` on a cluster) is only supported in
+    /// MongoDB 4.0 or greater.
+    ///
+    /// See the documentation [here](https://www.mongodb.com/docs/manual/changeStreams/) on change
+    /// streams.
+    ///
+    /// Change streams require either a "majority" read concern or no read
+    /// concern. Anything else will cause a server error.
+    pub fn watch(&self) -> Watch {
+        self.async_client.watch()
+    }
+}
+
+#[cfg(any(feature = "sync", feature = "tokio-sync"))]
+impl crate::sync::Database {
+    /// Starts a new [`ChangeStream`](change_stream/struct.ChangeStream.html) that receives events
+    /// for all changes in this database. The stream does not observe changes from system
+    /// collections and cannot be started on "config", "local" or "admin" databases.
+    ///
+    /// See the documentation [here](https://www.mongodb.com/docs/manual/changeStreams/) on change
+    /// streams.
+    ///
+    /// Change streams require either a "majority" read concern or no read
+    /// concern. Anything else will cause a server error.
+    pub fn watch(&self) -> Watch {
+        self.async_database.watch()
+    }
+}
+
+#[cfg(any(feature = "sync", feature = "tokio-sync"))]
+impl<T> crate::sync::Collection<T> {
+    /// Starts a new [`ChangeStream`](change_stream/struct.ChangeStream.html) that receives events
+    /// for all changes in this collection. A
+    /// [`ChangeStream`](change_stream/struct.ChangeStream.html) cannot be started on system
+    /// collections.
+    ///
+    /// See the documentation [here](https://www.mongodb.com/docs/manual/changeStreams/) on change
+    /// streams.
+    ///
+    /// Change streams require either a "majority" read concern or no read concern. Anything else
+    /// will cause a server error.
+    pub fn watch(&self) -> Watch {
+        self.async_collection.watch()
+    }
+}
+
+mod private {
+    pub trait Sealed {}
+    impl Sealed for super::Implicit {}
+    impl<'a> Sealed for super::Explicit<'a> {}
+}
+
+pub trait Session: private::Sealed {}
```

Comment:
I expect this will be needed for other methods that have a different return type when a session is present; when I get to that point I'll bump this up to the parent `action` module.